cmd/snap/cmd_run.go

```diff
@@ -85,7 +88,25 @@ func snapExecAppEnv(app *snap.AppInfo) []string {
 	return env
 }
 
+func createUserDataDirs(info *snap.Info) error {
+	usr, err := UserCurrent()
+	if err != nil {
+		return fmt.Errorf("cannot get the current user: %s", err)
+	}
+
+	// see snapenv.User
+	userData := filepath.Join(usr.HomeDir, info.MountDir())
+	commonUserData := filepath.Join(userData, "..", "common")
+	for _, d := range []string{userData, commonUserData} {
+		if err := os.MkdirAll(d, 0755); err != nil {
+			return fmt.Errorf("cannot create %q: %s", d, err)
+		}
+	}
+	return nil
+}
+
 var SyscallExec = syscall.Exec
+var UserCurrent = user.Current
```

Comment:
These should go into export_test.go as well. Their only reason of being here is tests, I think.